### DISCUSSION TITLE
How do you update your local repo while keeping config files?

### DISCUSSION THREAD

**Discussion by Michoko92** [hlky/stable-diffusion-webui#985] [Q&A] [answered]
Hi,

Sorry if this is a noob question, but I'm still learning all this git stuff. :)

Right now, every time I make a "git pull" of the repo to my local directory, Git tells me that there will be conflicts because I modified some files (mainly config files or options in relauncher.py). So I have to "git restore" those files, make a "git pull" to get latest updates, then modify again the files that need parameters change.

So I was wondering if there was an easier workflow with a specific Git command, where I could pull directly the latest version from the repo while keeping modifications made locally (with a possible automatic merge)?

Thank you!

**Comment by oc013** [comment, marked as answer]
You could try stashing your local changes which will bring it back to a "clean" state and let you pull fine, but you'll still possibly need to review for conflicts afterwards:
```
git stash
git pull
git stash pop
```

**Comment by Michoko92** [comment]
It worked, thank you very much for your help! :)

**Comment by outlookhazy** [comment]
Another workflow I use is to have a "personal" branch which has the changes I've made. When main gets updated, I re-base my branch on the latest main
